### DISCUSSION TITLE
Unreachable code detection based on Exception inheritance

### DISCUSSION THREAD

**Discussion by kumaraditya303** [microsoft/pyright#2595] [Ideas]
Take this code as an example:
```py
class Foo(Exception):
    pass


class FooBar(Foo):
    pass


def func() -> None:
    try:
        raise FooBar("foo")
    except Foo:
        print("Foo")
    except FooBar: # <--- This will never execute because FooBar is a subclass of Foo
        print("FooBar")

func()
```
In this example the FooBar will never be reachable as FooBar is a subclass of Foo and Foo already catches exception.

Pyright can flag this as unreachable.

**Comment by erictraut** [comment]
Thanks for the suggestion. I agree this pattern could be indicative of a bug. It could also be a harmless extra piece of code that was left in place. Let's see if other pyright users want to see a check for this condition. Please upvote the suggestion if you do.

**Comment by kumaraditya303** [comment]
For reference Python builtin exception tree https://docs.python.org/3/library/exceptions.html#exception-hierarchy
